celery/apps/beat.py

```diff
@@ -36,16 +41,16 @@ class Beat:
     """Beat as a service."""
 
     Service = beat.Service
-    app = None
-
-    def __init__(self, max_interval=None, app=None,
-                 socket_timeout=30, pidfile=None, no_color=None,
-                 loglevel='WARN', logfile=None, schedule=None,
-                 scheduler=None,
-                 scheduler_cls=None,  # XXX use scheduler
-                 redirect_stdouts=None,
-                 redirect_stdouts_level=None,
-                 quiet=False, **kwargs):
+    app: Celery
```

Comment:
```suggestion
    app: Celery = None
```

Reply:
I need to know, why should be remove app = None while trying to add type hints

Reply:
The reason why we need to remove `None` is to make the methods in the class type safe for mypy. As if it can be `None` here, and it can be `None` in the `init` method, then you can't call a single method on app without first checking if exists

Reply:
App gets overridden anyway when grabbing it from the app. If it isn't set, and no app was provided for the init, then it doesn't make a difference since it would have run into an attribute error trying to call a method on app anyway in the `init` method anway.